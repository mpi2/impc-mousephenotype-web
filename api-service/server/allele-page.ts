import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { AlleleSummary } from "@/models";
const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchAlleleSummary(
  mgiGeneAccessionId: string,
  alleleName: string,
): Promise<AlleleSummary> {
  const endpointURL = `http://impc-allele-service.${KUBERNETES_NAMESPACE}:8080/v1/alleles?mgiGeneAccessionId=${mgiGeneAccessionId}&alleleName=${alleleName}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<AlleleSummary>(
        `/api/v1/alleles/${mgiGeneAccessionId}/${alleleName}`,
      )
    : fetchURL<AlleleSummary>(endpointURL));
}
