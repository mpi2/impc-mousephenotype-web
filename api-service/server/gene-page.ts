import { GeneOrder, GenePhenotypeHits, GeneSummary } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { processGeneOrderResponse } from "@/hooks/gene-order.query";
import { processGenePhenotypeHitsResponse } from "@/hooks/significant-phenotypes.query";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchGeneSummary(
  mgiGeneAccessionId: string,
): Promise<GeneSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/gene_summary?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<GeneSummary>(
        `/api/v1/genes/${mgiGeneAccessionId}/summary`,
      )
    : fetchURL<GeneSummary>(endpointURL));
}

export async function fetchGenePhenotypeHits(
  mgiGeneAccessionId: string,
): Promise<Array<GenePhenotypeHits>> {
  const endpointURL = `http://impc-phenotype-hits-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotypehits?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GenePhenotypeHits>>(
        `/api/v1/genes/${mgiGeneAccessionId}/phenotype-hits`,
      )
    : fetchURL<Array<GenePhenotypeHits>>(endpointURL));
  return processGenePhenotypeHitsResponse(data);
}

export async function fetchGeneOrderData(
  mgiGeneAccessionId: string,
): Promise<Array<GeneOrder>> {
  const endpointURL = `http://impc-order-service.${KUBERNETES_NAMESPACE}:8080/v1/orders?mgiGeneAccessionId=${mgiGeneAccessionId}`;
  const data = await (NODE_ENV === "development"
    ? fetchAPIFromServer<Array<GeneOrder>>(
        `/api/v1/genes/${mgiGeneAccessionId}/order`,
      )
    : fetchURL<Array<GeneOrder>>(endpointURL));
  return processGeneOrderResponse(data);
}
