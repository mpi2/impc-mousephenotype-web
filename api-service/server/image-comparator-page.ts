import { GeneImageCollection } from "@/models/gene";
import { fetchAPIFromServer, fetchURL } from "@/api-service";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchMutantImages(
  mgiGeneAccessionId: string,
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByMgiAndStableId?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_mgi_and_stable_id?mgiGeneAccessionId=${mgiGeneAccessionId}&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}

export async function fetchControlImages(
  parameterStableId: string,
): Promise<Array<GeneImageCollection>> {
  const endpointURL = `http://impc-imaging-service.${KUBERNETES_NAMESPACE}:8080/v1/images/findByStableIdAndSampleId?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`;
  try {
    return await (NODE_ENV === "development"
      ? fetchAPIFromServer<Array<GeneImageCollection>>(
          `/api/v1/images/find_by_stable_id_and_sample_id?biologicalSampleGroup=control&parameterStableId=${parameterStableId}`,
        )
      : fetchURL<Array<GeneImageCollection>>(endpointURL));
  } catch (error) {
    return [];
  }
}
