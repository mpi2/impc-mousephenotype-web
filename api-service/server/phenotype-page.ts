import { fetchAPIFromServer, fetchURL } from "@/api-service";
import { PhenotypeSummary } from "@/models/phenotype";

const KUBERNETES_NAMESPACE = process.env.KUBERNETES_NAMESPACE ?? "default";
const NODE_ENV = process.env.NODE_ENV;

export async function fetchPhenotypeSummary(
  phenotypeId: string,
): Promise<PhenotypeSummary> {
  const endpointURL = `http://impc-summary-service.${KUBERNETES_NAMESPACE}:8080/v1/phenotype_summary?phenotypeId=${phenotypeId}`;
  return await (NODE_ENV === "development"
    ? fetchAPIFromServer<PhenotypeSummary>(
        `/api/v1/phenotypes/${phenotypeId}/summary`,
      )
    : fetchURL<PhenotypeSummary>(endpointURL));
}
