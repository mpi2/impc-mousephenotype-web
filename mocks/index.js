export async function initMocks() {
  if (typeof window === "undefined") {
    const { server } = await import("./server");
    server.listen();
  } else {
    const { worker } = await import("./browser");
    return worker.start({
      serviceWorker: {
        url: "/data/mockServiceWorker.js",
        options: {
          scope: "/",
        }
      }
    });
  }
}

