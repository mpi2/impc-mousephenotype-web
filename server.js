const { createServer } = require('http');
const { parse } = require('url');
const next = require('next');

const dev = process.env.NODE_ENV !== 'production';
const hostname = 'localhost';
const port = 3000;
const app = next({ dev, hostname, port });
const handle = app.getRequestHandler();

app.prepare()
.then(() => {
    createServer(async (req, res) => {
        try {
            const parsedUrl = parse(req.url, true);
            const { pathname, query} = parsedUrl;
            const host = req.headers.host;
            console.log({ host, pathname });
            res.setHeader('X-IMPC-TEST', "true");
            if (host.includes('nginx.mousephenotype-dev.org')) {
                const modifiedPathName = pathname.replace("/data/", "");
                console.log({ modifiedPathName });
                await app.render(req, res, modifiedPathName, query);
                res.setHeader('X-PATH', JSON.stringify({ host, pathname, modifiedPathName }));
            } else {
                await handle(req, res, parsedUrl);
            }

        } catch (err) {
            console.error('Error occurred handling', req.url, err)
            res.statusCode = 500
            res.end('internal server error')
        }
    })
        .once('error', (err) => {
            console.error(err);
            process.exit(1);
        })
        .listen(port, () => {
            console.log(`Ready on http://${hostname}:${port}`);
        })
});