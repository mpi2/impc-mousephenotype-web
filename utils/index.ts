import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};
