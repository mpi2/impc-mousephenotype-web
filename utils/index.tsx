import React from "react";
import _ from "lodash";

export const formatBodySystems = (systems: string[] | string = []) => {
  return _.capitalize(
    (typeof systems === "string" ? systems : systems.join(", "))
      .replace(/ phenotype/g, "")
      .replace(/\//g, " / ")
  );
};

export const formatAlleleSymbol = (allele: string) => {
  return allele.slice(0, allele.length - 1).split("<");
};

export const formatPValue = (pValue: number) => {
  const pValueArray = Number.parseFloat(String(pValue))
    .toExponential(2)
    .split("e");
  return (
    <>
      {pValueArray[0]}x10<sup>{pValueArray[1].replace("+", "")}</sup>
    </>
  );
};

export const formatESCellName = (src: string) => {
  const [name, superscript, end] = src.split(/<|>/);
  return (
    <>
      {name}
      <sup>{superscript}</sup>
      {end}
    </>
  );
};

const fetchCache = {};
export { fetchCache };

export const toSentenceCase = (camelCase) => {
  if (camelCase) {
    const result = camelCase.replace(/([A-Z])/g, " $1");
    return result[0].toUpperCase() + result.substring(1).toLowerCase();
  }
  return "";
};
