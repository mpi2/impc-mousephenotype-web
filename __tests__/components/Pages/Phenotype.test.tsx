import { screen } from "@testing-library/react";
import PhenotypePage from "@/app/phenotypes/[id]/phenotype-page";
import { API_URL, renderWithClient } from "../../utils";
import { testServer } from "../../../mocks/server";
import { rest } from "msw";
import phenotypeData from "../../../mocks/data/phenotypes/MP:0012361/summary.json";
import phenotypeStatsData from "../../../mocks/data/phenotypes/MP:0012361/phenotype-stats-results.json";
import phenotypeGenotypeHitsData from "../../../mocks/data/phenotypes/MP:0012361/genotype-hits.json";

window.ResizeObserver =
  window.ResizeObserver ||
  jest.fn().mockImplementation(() => ({
    disconnect: jest.fn(),
    observe: jest.fn(),
    unobserve: jest.fn(),
  }));

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useSearchParams: jest.fn().mockImplementation(() => new URLSearchParams()),
  usePathname: jest.fn(),
}));

describe("Phenotype page", () => {
  testServer.use(
    rest.get(
      `${API_URL}/api/v1/phenotypes/MP:0012361/genotype-hits/by-any-phenotype-Id`,
      (req, res, ctx) => {
        return res(ctx.json(phenotypeGenotypeHitsData));
      },
    ),
    rest.get(
      `https://impc-datasets.s3.eu-west-2.amazonaws.com/phenotype-stats-results/dr22.1/MP:0012361.json`,
      (req, res, ctx) => {
        return res(ctx.json(phenotypeStatsData));
      },
    ),
  );

  it("renders correctly", async () => {
    const { container } = renderWithClient(
      <PhenotypePage
        phenotypeId="MP:0012361"
        phenotype={phenotypeData}
        phenotypeHits={[]}
      />,
    );
    expect(await screen.findByLabelText("Manhattan Plot")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
