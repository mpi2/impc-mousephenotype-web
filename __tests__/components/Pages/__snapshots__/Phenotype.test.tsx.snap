// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phenotype page renders correctly 1`] = `
<div>
  <div
    class="banner"
  >
    <div
      class="pb-4 pt-5 container container"
    >
      <div
        class="col-12 col-md-8 ps-4 pe-4"
      >
        <div
          class="tabs"
        >
          <a
            class="tab"
            href="/search"
          >
            Genes
          </a>
          <a
            class="tab__active"
            href="/search?type=phenotype"
          >
            Phenotypes
          </a>
          <a
            class="tab"
            href="https://dev.mousephenotype.org/?s="
          >
            Help, news, blog
          </a>
        </div>
        <div
          class="inputCont"
        >
          <input
            class="input"
            placeholder="Search for a phenotype..."
            title="main search box"
            type="text"
            value=""
          />
          <button
            aria-describedby="svg-inline--fa-title-search-icon"
            class="searchBtn"
          >
            <svg
              aria-labelledby="svg-inline--fa-title-search-icon"
              class="svg-inline--fa fa-magnifying-glass "
              data-icon="magnifying-glass"
              data-prefix="fas"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title
                id="svg-inline--fa-title-search-icon"
              >
                Search button
              </title>
              <path
                d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="page container"
  >
    <div
      class="card"
    >
      <div
        class="subheadingCont"
      >
        <div
          class="subheading"
        >
          <span
            class="subheadingSection"
          >
            Phenotype
          </span>
          <span
            class="subheadingSection"
          />
        </div>
      </div>
      <br />
      <p
        class="grey"
      >
        Loading...
      </p>
    </div>
    <div
      class="card"
      id="associations-table"
    >
      <h2>
        IMPC Gene variants with 
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton"
            style="width: 150px;"
          >
            ‌
          </span>
        </span>
      </h2>
      <p>
        Total number of significant genotype-phenotype associations: 
        0
      </p>
      <div
        class="buttonsWrapper top withControls"
        data-testid="top-controls-wrapper"
      >
        <div
          class="additionalWrapper"
        >
          <input
            aria-label="Filter by parameters"
            class="bg-white form-control"
            id="parameterFilter"
            placeholder="Search "
            style="display: inline-block; width: 200px; margin-right: 2rem;"
            type="text"
          />
        </div>
      </div>
      <div
        class="tableWrapper"
      >
        <table
          class="table striped undefined table table-bordered"
        >
          <thead>
            <tr>
              <th
                rowspan="2"
                width="16.666666666666664%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: bold;"
                >
                  Gene / allele
                   
                  <svg
                    aria-hidden="true"
                    class="svg-inline--fa fa-caret-up sortIcon"
                    data-icon="caret-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                      fill="currentColor"
                    />
                  </svg>
                </button>
              </th>
              <th
                rowspan="2"
                width="10.833333333333334%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Phenotype
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Supporting data
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Zygosity
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="5.833333333333333%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Sex
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Life stage
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Parameter
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="12.5%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Phenotyping Center
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
              <th
                rowspan="2"
                width="8.333333333333332%"
              >
                <button
                  class="inlineButton headerButton"
                  style="font-weight: inherit;"
                >
                  Most significant P-value
                   
                  <span
                    class="defaultIcons"
                  >
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-up first"
                      data-icon="caret-up"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M182.6 137.4c-12.5-12.5-32.8-12.5-45.3 0l-128 128c-9.2 9.2-11.9 22.9-6.9 34.9s16.6 19.8 29.6 19.8H288c12.9 0 24.6-7.8 29.6-19.8s2.2-25.7-6.9-34.9l-128-128z"
                        fill="currentColor"
                      />
                    </svg>
                    <svg
                      aria-hidden="true"
                      class="svg-inline--fa fa-caret-down second"
                      data-icon="caret-down"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      viewBox="0 0 320 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M137.4 374.6c12.5 12.5 32.8 12.5 45.3 0l128-128c9.2-9.2 11.9-22.9 6.9-34.9s-16.6-19.8-29.6-19.8L32 192c-12.9 0-24.6 7.8-29.6 19.8s-2.2 25.7 6.9 34.9l128 128z"
                        fill="currentColor"
                      />
                    </svg>
                  </span>
                </button>
              </th>
            </tr>
            <tr />
          </thead>
          <tbody>
            <tr>
              <td
                colspan="9"
              >
                <b>
                  No data available
                </b>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="buttonsWrapper withControls"
      >
        <div
          class="additionalWrapper bottomControls"
        >
          <div
            class="grey"
            style="display: flex; gap: 0.5rem; align-items: center;"
          >
            Download data as:
             
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               TSV
            </button>
             
            <button
              class="btn impc-secondary-button small"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-download fa-sm "
                data-icon="download"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32V274.7l-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7V32zM64 352c-35.3 0-64 28.7-64 64v32c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V416c0-35.3-28.7-64-64-64H346.5l-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352H64zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
                  fill="currentColor"
                />
              </svg>
               XLS
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <h2>
        Most significant associations for 
      </h2>
      <div
        class="mainWrapper"
      >
        <div
          class="chart"
        >
          <div
            class="labelsWrapper"
          >
            <div
              style="font-size: 90%;"
            >
              <i
                class="fa fa-circle"
                style="color: rgb(212, 17, 89);"
              />
                Not significant
              <i
                class="fa fa-circle"
                style="color: rgb(26, 133, 255); margin-left: 1rem;"
              />
                Significant
              <i
                class="fa fa-circle"
                style="color: rgb(0, 159, 129); margin-left: 1rem;"
              />
                Manual associations
              <div
                style="display: inline-block;"
              >
                <hr
                  style="border-top: 3px dashed #000; height: 3px; width: 50px; display: inline-block; margin: 0px 0px 0px 0.5rem; opacity: 1;"
                />
                  Significant P-value threshold (P &lt; 0.0001)
              </div>
            </div>
            <div
              style="display: flex; white-space: nowrap; align-items: center;"
            >
              <label
                class="grey"
                for="geneHighlight"
                style="margin-right: 0.5rem;"
              >
                Find gene:
              </label>
              <input
                class="form-control"
                id="geneHighlight"
                type="text"
                value=""
              />
            </div>
          </div>
          <i
            class="grey"
            style="padding: 0px 2rem;"
          >
            Associations appearing in the region of 1x10
            <sup>
              -30
            </sup>
             are manually annotated as significant.
          </i>
          <div
            class="mt-4"
            style="display: flex; justify-content: center;"
          >
            <div
              data-test-id="CircularProgressbarWithChildren"
            >
              <div
                style="position: relative; width: 100%; height: 100%;"
              >
                <svg
                  class="CircularProgressbar rotating"
                  data-test-id="CircularProgressbar"
                  viewBox="0 0 100 100"
                >
                  <path
                    class="CircularProgressbar-trail"
                    d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
                    fill-opacity="0"
                    stroke-width="12"
                    style="stroke: #fff; stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 0px;"
                  />
                  <path
                    class="CircularProgressbar-path"
                    d="
      M 50,50
      m 0,-44
      a 44,44 0 1 1 0,88
      a 44,44 0 1 1 0,-88
    "
                    fill-opacity="0"
                    stroke-width="12"
                    style="stroke: #ccc; transition: stroke-dashoffset 2s ease 0s; stroke-dasharray: 276.46015351590177px 276.46015351590177px; stroke-dashoffset: 55.29203070318034px;"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
        <div
          class="resultsSection"
        >
          <h3>
            Associations details of selected gene(s):
          </h3>
          <i
            class="grey"
          >
            Click on a point of the chart to see the details of an association.
          </i>
        </div>
      </div>
    </div>
    <div
      class="card"
    >
      <h2>
        The way we measure
      </h2>
      <p>
        Procedure
      </p>
    </div>
  </div>
</div>
`;
